exonum/src/blockchain/tests.rs

```diff
@@ -273,15 +272,109 @@ fn handling_tx_panic_storage_error(blockchain: &Blockchain) {
     );
 }
 
+struct ServiceGood;
+
+impl Service for ServiceGood {
+    fn service_id(&self) -> u16 {
+        1
+    }
+
+    fn service_name(&self) -> &'static str {
+        "some_service"
+    }
+
+    fn state_hash(&self, _snapshot: &Snapshot) -> Vec<Hash> {
+        vec![]
+    }
+
+    fn tx_from_raw(&self, _raw: RawTransaction) -> Result<Box<Transaction>, MessageError> {
+        unimplemented!()
+    }
+
+    fn handle_execute(&self, fork: &mut Fork) {
+        let mut index = ListIndex::new(IDX_NAME, fork);
+        index.push(1);
+    }
+}
+
+struct ServicePanic;
+
+impl Service for ServicePanic {
+    fn service_id(&self) -> u16 {
+        1
+    }
+
+    fn service_name(&self) -> &'static str {
+        "some_service"
+    }
+
+    fn state_hash(&self, _snapshot: &Snapshot) -> Vec<Hash> {
+        vec![]
+    }
+
+    fn tx_from_raw(&self, _raw: RawTransaction) -> Result<Box<Transaction>, MessageError> {
+        unimplemented!()
+    }
+
+    fn handle_execute(&self, _fork: &mut Fork) {
+        panic!("42");
+    }
+}
+
+struct ServicePanicStorageError;
+
+impl Service for ServicePanicStorageError {
+    fn service_id(&self) -> u16 {
+        1
+    }
+
+    fn service_name(&self) -> &'static str {
+        "some_service"
+    }
+
+    fn state_hash(&self, _snapshot: &Snapshot) -> Vec<Hash> {
+        vec![]
+    }
+
+    fn tx_from_raw(&self, _raw: RawTransaction) -> Result<Box<Transaction>, MessageError> {
+        unimplemented!()
+    }
+
+    fn handle_execute(&self, _fork: &mut Fork) {
+        panic!(Error::new("42"));
+    }
+}
+
+fn handling_handle_execute(blockchain: &Blockchain, db: &mut Box<Database>) {
```

Comment:
Why are these functions called `handling*`? IMO, `assert*` would be more appropriate.